lib/enumerable/inject.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_inject(initial = nil)
+    raise 'Please provide a block' unless block_given?
+
```

Comment:
- [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
- [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

Reply:
```
return to_enum unless block_given?

    result = initial.nil? ? self[0] : initial

    [*1..length - 1].my_each do |i|
      result = yield(result, self[i])
    end
    result
```

Solved!!

Reply:
Added more cases, solved